text/0000-offline-buildpackages.md

```diff
@@ -0,0 +1,221 @@
+# Meta
+[meta]: #meta
+- Name: Offline Buildpackages
+- Start Date: 2020-04-27
+- Author(s): dwillist
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: NA
+
+# Summary
+[summary]: #summary
+
+This change allows buildpacks to access vendored assets during builds to enable offline builds and asset reuse.
+
+To do so we introduce the asset package, a distribution artifact that provides vendored assets for buildpacks, and platform mechanisms to provide vendored assets to buildpacks during `build` and `detect`.
+
+# Motivation
+[motivation]: #motivation
+
+- Simplify artifacts needed to achieve offline builds.
+- Enable registry deduplication of buildpack assets
+- Separate buildpacks from vendored assets.
+
+# What it is
+[what-it-is]: #what-it-is
+
+### Define the target persona:
+Buildpack user, Platform operator, Buildpack author
+
+### Explaining the feature largely in terms of examples.
+We define the asset package. Which is a reproducible OCI image. Each layer in this image contains one or more `asset` files that a buildpack may want to contribute during a build. Each of these `asset` files will be made available at `/cnb/assets/<asset-sha256>.extension` in the build container.
+
+Asset Package Layers Layout:
+```
+<layer1> ┳━ /cnb/assets/<java11-asset-sha256>
+         ┗━ /cnb/assets/<java13-asset-sha256>
+
+...
+<layern> ━━ /cnb/assets/<java15-asset-sha256>
+```
+
+Asset packages will have a `io.buildpacks.asset.layers` Label. This label's contents will be a `json` object mapping each layer to metadata describing the assets it provides.
+
+``` json
+{
+  "<layer1-diffID>": [
+    {
+      "sha256": "java11-asset-sha256",
+      "uri": "https://path/to/java11.tgz",
```

Comment:
Can you clarify what `uri` is for? The package already contains the data, is it there for audit purposes?

Reply:
Yep, it just there so that we have some human readable information about what is in the layer.